### ISSUE TITLE
error_with_backtrace_outputs_correctly_with_one_source fails on Windows7

### ISSUE THREAD

**Issue by roblabla** [rust-lang/rust#117941]
Since https://github.com/rust-lang/rust/pull/117089 was merged, the `error_with_backtrace_outputs_correctly_with_one_source` std test fails with the following error on windows7:

```
---- error::tests::error_with_backtrace_outputs_correctly_with_one_source stdout ----
thread 'error::tests::error_with_backtrace_outputs_correctly_with_one_source' panicked at library/std/src/../../backtrace/src/dbghelp.rs:168:1:
called `Option::unwrap()` on a `None` value
```

After some further instrumentation, I was able to find out the source of the problem: `Failed to find SymAddrIncludeInlineTrace in dbghelp`. This was likely caused by https://github.com/rust-lang/backtrace-rs/commit/c1464fadd7c870c87818b86b6161ee188be6f3a4 , which reworked the backtracing code on windows to be faster through the use of newer APIs. Unfortunately, not all of those APIs are available in Windows 7.

### Version it worked on

72012402e1448348b8adf9423647c18a2f094726

### Version with regression

608e9682f0a6482903de8d3332770104a0ad943c

@rustbot modify labels: +regression-from-stable-to-nightly -regression-untriaged

**Comment by lqd** [comment]
cc @wesleywiser #117089 is going to land on 1.75 while https://github.com/rust-lang/compiler-team/issues/651 drops windows <10 support from 1.76

**Comment by roblabla** [comment]
@lqd While this manifests as a test error, I'm pretty sure there's an actual regression here that can be observed in code, and isn't just a unit test failure. I haven't taken the time to make an actual reproducer, but I think any attempt to print a `Backtrace` through its `Display` implementation will fail. I'll try to make some code showing the regression tomorrow.

Regarding windows7 being not well-supported, I'm well-aware of that. I'm trying to improve on this by running the library/std tests on windows7 every night, and eventually adding a new (tier3) windows7 target once support is officially dropped in 1.76 so we can continue targeting it.

**Comment by lqd** [comment]
I agree this is likely a regression, and pinged the author: it’s likely the fact that the PR will not work on windows 7 wasn’t known then, but at the same time it seems it could be accepted if it landed today.

**Comment by apiraino** [comment]
> Unfortunately, not all of those APIs are available in Windows 7.

@roblabla Just to be sure I understand: Do I read correctly that due to the absence of some APIs, the issue won't reproduce on Win10?

WG-prioritization assigning priority ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.23117941.20error_with_backtrace_outputs_correctly_with_one_sou.E2.80.A6)).

@rustbot label -I-prioritize +P-low

**Comment by roblabla** [comment]
I just realized this actually affects the 1.75 beta branch as well, this is a regression from stable to beta.

**Comment by ChrisDenton** [comment]
Once the backtrace PR is merged another PR updating the crate can be merged here with the `beta-nominated` label.
